code/modules/wiremod/components/atom/remotecam.dm

```diff
@@ -0,0 +1,454 @@
+
+	camera_range_settable = 0
+
+	current_camera_range = 0
+
+	var/obj/machinery/door/airlock/airlock = null
+
+/obj/item/circuit_component/compare/remotecam/polaroid
+	display_name = "Polaroid Camera Add-On"
+	desc = "Relays a polaroid camera's feed as a digital stream for surveillance-on-the-go. Network field is used for camera network."
+
+	required_shells = list(/obj/item/camera)
+
+	camera_range_settable = 0
+
+	current_camera_range = 0
+
+	var/obj/item/circuit_component/camera/polaroid = null
+
+
+/obj/item/circuit_component/compare/remotecam/bci/input_received(datum/port/input/port)
+	if(bci && shell_camera)
+		update_camera_process(port, "BCI")
+	//Do not update output ports if changed network or camera range
+	if(port != network && port != camera_range)
+		. = ..()
+
+/obj/item/circuit_component/compare/remotecam/drone/input_received(datum/port/input/port)
+	if(drone && shell_camera)
+		update_camera_process(port, "Drone")
+	//Do not update output ports if changed network or camera range
+	if(port != network && port != camera_range)
+		. = ..()
+
+/obj/item/circuit_component/compare/remotecam/airlock/input_received(datum/port/input/port)
+	if(airlock && shell_camera)
+		update_camera_process(port, "Airlock")
+	//Do not update output ports if changed network
+	if(port != network)
+		. = ..()
+
+/obj/item/circuit_component/compare/remotecam/polaroid/input_received(datum/port/input/port)
+	if(polaroid && shell_camera)
+		update_camera_process(port, "Polaroid")
+	//Do not update output ports if changed network
+	if(port != network)
+		. = ..()
+
+/obj/item/circuit_component/compare/remotecam/bci/Destroy()
+	bci = null
+	return ..()
+
+/obj/item/circuit_component/compare/remotecam/drone/Destroy()
+	drone = null
+	return ..()
+
+/obj/item/circuit_component/compare/remotecam/airlock/Destroy()
+	airlock = null
+	return ..()
+
+/obj/item/circuit_component/compare/remotecam/polaroid/Destroy()
+	polaroid = null
+	return ..()
+
+/obj/item/circuit_component/compare/remotecam/bci/register_shell(atom/movable/shell)
+	. = ..()
+	bci = null
+	shell_camera = null
+	if(istype(shell, /obj/item/organ/internal/cyberimp/bci))
+		bci = shell
+		shell_camera = new /obj/machinery/camera (bci)
+		init_camera("BCI")
+
+/obj/item/circuit_component/compare/remotecam/drone/register_shell(atom/movable/shell)
+	. = ..()
+	drone = null
+	shell_camera = null
+	if(istype(shell, /mob/living/circuit_drone))
+		drone = shell
+		shell_camera = new /obj/machinery/camera (drone)
+		init_camera("Drone")
+
+/obj/item/circuit_component/compare/remotecam/airlock/register_shell(atom/movable/shell)
+	. = ..()
+	airlock = null
+	shell_camera = null
+	if(istype(shell, /obj/machinery/door/airlock))
+		airlock = shell
+		shell_camera = new /obj/machinery/camera (airlock)
+		init_camera("Airlock")
+
+/obj/item/circuit_component/compare/remotecam/polaroid/register_shell(atom/movable/shell)
+	. = ..()
+	polaroid = null
+	shell_camera = null
```

Comment:
these should not be nulled out when `register_shell` is called.
Your `unregister_shell` function should be doing this instead.